azurerm/resource_arm_firewall.go

```diff
@@ -407,3 +413,35 @@ func validateAzureFirewallSubnetName(v interface{}, k string) (warnings []string
 
 	return warnings, errors
 }
+
+func validateFirewallConfigurationSettings(d *schema.ResourceData) error {
+	configs := d.Get("ip_configuration").([]interface{})
+	subnetNumber := 0
+	hasInternal := false
+	hasPublic := false
+
+	for _, configRaw := range configs {
+		data := configRaw.(map[string]interface{})
+		if subnet, exist := data["subnet_id"].(string); exist && subnet != "" {
+			subnetNumber++
+		}
```

Comment:
I'm not sure what the purpose of this is.. I think this can be removed?

Reply:
For subnetNumber:

The constraint is that only one subnet_id is required and allowed among all the elements of ip_configuration ( Azure RestApi will throw an error if multiple subnet_id exist or there is no subnet_id)